tests/RobotFramework/tests/cumulocity/software_management/software.robot

```diff
@@ -9,11 +9,28 @@ Test Teardown    Custom Teardown
 
 *** Test Cases ***
 Supported software types should be declared during startup
-    [Documentation]    #2654 This test will be updated once advanced software management support is implemented
+    [Documentation]    c8y_SupportedSoftwareTypes should NOT be created by default #2654
     Should Have MQTT Messages    topic=te/device/main///cmd/software_list    minimum=1    maximum=1    message_contains="types":["apt"]
     Should Have MQTT Messages    topic=te/device/main///cmd/software_update    minimum=1    maximum=1    message_contains="types":["apt"]
+    Run Keyword And Expect Error    *    Device Should Have Fragment Values    c8y_SupportedSoftwareTypes\=["apt"]
+
+Supported software types and c8y_SupportedSoftwareTypes should be declared during startup
+    [Documentation]    c8y_SupportedSoftwareTypes should be created if the relevant config is set to true #2654
+    Execute Command    tedge config set c8y.software_management.with_types true
+    Restart Service    tedge-mapper-c8y
```

Comment:
After restarting service, isn't it better to add some sleep time in order to allow the service to be up again?

Reply:
No need to put a sleep here as the `Device Should Have Fragments value` is a smart assertion which will retry automatically until a timeout period.